cep-XXXX.md

```diff
@@ -0,0 +1,119 @@
+
+#### Channel base URLs and names
+
+The base URL for the arbitrary location of a repodata file is defined as:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]/<subdir>/repodata.json
+```
+
+with `<scheme>`, `<authority>` and `<path>` defined by [RFC
+3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2).
+
+Taken the channel definition above, the base URL without trailing slashes is thus:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]
+```
+
+For example, given `https://conda.anaconda.org/conda-forge/noarch/repodata.json`, the part leading
+to `noarch/repodata.json` and thus base URL is `https://conda.anaconda.org/conda-forge/conda-forge`. For local repodata such as `file:///home/username/channel/noarch/repodata.json`, the
+channel base URL is `file:///home/username/channel`.
+
+For convenience, the channel _name_ is defined as the concatenation of `scheme`, `authority` and
+`path` components. `scheme` MAY be omitted for `http`, `https` and `file`. At least one of
+`authority` or `path` SHOULD be present. In their absence, the channel name MUST be considered
+empty, regardless the scheme. Empty channel names SHOULD NOT be used.
+
+When present, each path component SHOULD only contain lowercase ASCII letters, numbers, underscores
+and dashes. They SHOULD start and end with a letter or a number.
+
+The maximum length of a channel base URL SHOULD NOT exceed 256 characters.
+
+#### Subdir names
+
+Channel subdir names MUST either be the literal `noarch` or a string following the syntax `{os}-{arch}`, where `{os}` and `{arch}` MUST only consist of lowercase ASCII letters and numbers. Non-`noarch` subdirs MUST match this regex: `^[a-z0-9]+-[a-z0-9]+$`.
+
+The maximum length of a subdir name MUST NOT exceed 32 characters.
+
+#### Label names
+
+Channel label names MUST only consist of ASCII letters, digits, underscores, hyphens, forward slashes, periods, colons and whitespace. They MUST start with a letter. They MUST match this regex: `^[a-zA-Z][0-9a-zA-Z_\-\.\/:\s]*`. Even if allowed, label names SHOULD NOT contain any whitespace.
+
+The label `nolabel` is reserved and MUST only be used for conda packages which have no other labels. In other words, in the space of labels, the empty set is represented by the labels `nolabel`.
+
+The maximum length of a label name SHOULD NOT exceed 128 characters.
+
+## Backwards compatibility
+
+The conda channel, subdir, label, and package name regexes are backwards compatible with the current `conda` implementation (25.3) and all existing packages on the `defaults` and `conda-forge` channels, except the `__anaconda_core_depends` package on the `defaults` channel.
+
+The regex for labels above was pulled from an anaconda.org error message describing the set of valid labels. Finally, anaconda.org usernames/channel names are case insensitive and cannot begin with an underscore.
```

Comment:
It would be good to collect the stats from @chenghlee on anaconda.org and put them as text here.